### DISCUSSION TITLE
Getting the available container size

### DISCUSSION THREAD

**Discussion by rgathmann** [QuestPDF/QuestPDF#36] [Q&A]
Is it possible to access the available Container size similar to how Canvas receives a size? Ultimately, I would like to be able to repeat an element to fill the available space, or perform other calculations against the container size. Thanks!

**Comment by MarcinZiabek** [comment]
Hello, This functionality is currently not available and was skipped as a design decision. I think that if you need the aforementioned functionality, the suite of available elements may be lacking something. Can you please provide me with a real example what would you like to accomplish so I better understand your requirements? This way we can validate if indeed it is not possible to do with the current library and then reevaluate available functionalities.

**Comment by MarcinZiabek** [comment]
@rgathmann Would you like to provide an example of a requested container? This will help me understand your needs :)

**Reply by rgathmann** [reply]
Sorry for my slow response! So I am working on generating flexible single page layouts, for example planners, calendars, etc. For these, its more important to consider the available space and then be able to make decisions about how large things should be inside of it instead of the way it is done today. For example, if I wanted a column that is 1/4 height notes, 1/2 schedule, and 1/4 checklist, there isn't a way to do this today.

That being said, I totally understand why the choices were made that promotes easy paging because they is probably the more common scenario. As a work around, I have been using the canvas to create the layouts that I want which works pretty well!

**Reply by MarcinZiabek** [reply]
I am currently working on some solutions that may be really useful in your use-cases. This, in a sense, is a hybrid between creating custom content and using the Canvas element. [In this discussion](https://github.com/QuestPDF/QuestPDF/discussions/39), I have been exploring various possibilities regarding dynamic component generation which should easily cover your scenarios. It still requires a lot of work inside the architecture, I hope it will be available in a 2022.01 January release :)

**Comment by MarcinZiabek** [comment]
@rgathmann Hi, you may be interested in the recent accomplishments [in this discussion](https://github.com/QuestPDF/QuestPDF/discussions/39) :)

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
